pkg/workflows/workload/writeclusterconfig.go

```diff
@@ -0,0 +1,46 @@
+package workload
+
+import (
+	"context"
+
+	"github.com/aws/eks-anywhere/pkg/clustermarshaller"
+	"github.com/aws/eks-anywhere/pkg/logger"
+	"github.com/aws/eks-anywhere/pkg/task"
+)
+
+type writeClusterConfig struct{}
+
+// Run writeClusterConfig writes new management cluster's cluster config file to the destination after the create/upgrade process.
+func (s *writeClusterConfig) Run(ctx context.Context, commandContext *task.CommandContext) task.Task {
+	logger.Info("Writing cluster config file")
+	err := clustermarshaller.WriteClusterConfig(commandContext.ClusterSpec, commandContext.Provider.DatacenterConfig(commandContext.ClusterSpec), commandContext.Provider.MachineConfigs(commandContext.ClusterSpec), commandContext.Writer)
+	if err != nil {
+		commandContext.SetError(err)
+	}
+
+	successMsg := ""
+	if commandContext.CurrentClusterSpec != nil {
+		successMsg = "Cluster upgraded!"
+	} else {
+		successMsg = "Cluster created!"
+	}
```

Comment:
the abstractions presented by `pkg/workflows` don't favor reusing tasks
the fact that you are tying to do this here seems to indicate maybe we should look into `pkg/workflow` instead. You don't need to do it, we can move forward with this PR as is. But maybe worth thinking about it

Reply:
Yeah, I can fix that in a future PR.